File: application/src/Caravelo/Bundle/ApiBundle/Model/SurveyDto.php
<?php
namespace Caravelo\Bundle\ApiBundle\Model;

use Caravelo\Modules\Survey\Domain\Model\Survey;

class SurveyDto
{
    public $name;

    public $slug;

    public $description;

    public $tags = [];

    public $questions = [];

    /**
     * @param Survey $survey
     * @return $this
     */
    public function parseSurvey(Survey $survey)
    {
        $this->name = $survey->getName();
        $this->slug = $survey->getSlug();
        $this->description = $survey->getDescription();

        foreach ($survey->getQuestions() as $question) {
            $questionDto = new QuestionDto();
            $questionDto->parseQuestion($question);
            $this->questions[] = $questionDto;
        }

        foreach ($survey->getTags() as $tag) {
            $tagDto = new TagDto();
            $tagDto->parseTag($tag);
            $this->tags[] = $tagDto;
        }

        return $this;
    }
}

File: application-java/src/main/java/iruiz/app/dto/SurveyDto.java
package iruiz.app.dto;

import java.util.ArrayList;

public class SurveyDto {
    public String name;

    public String slug;

    public String description;
    
    public String createdAt;

    public ArrayList<TagDto> tags;

    public ArrayList<QuestionDto> questions;
    
    public SurveyDto(){
    	this.questions = new ArrayList<QuestionDto>();
    	this.tags = new ArrayList<TagDto>();
    }
    
    public void addTag(TagDto tag){
    	this.tags.add(tag);
    }
    
    public void addQuestion(QuestionDto question){
    	this.questions.add(question);
    }
}
